File: java/amp-examples/stream-publisher/src/main/java/amp/examples/streaming/publisher/StreamingPublisher.java
package amp.examples.streaming.publisher;

import amp.eventing.streaming.DefaultStreamingBus;
import amp.examples.streaming.common.ModernMajorGeneralMessage;
import cmf.eventing.patterns.streaming.IEventStream;
import cmf.eventing.patterns.streaming.IStreamingEventBus;
import org.springframework.context.support.ClassPathXmlApplicationContext;

import java.util.ArrayList;

public class StreamingPublisher {

    /**
     * Demonstrates how to publish a chunked sequence of messages or using the eventStream to publish for you.
     * @param args
     * @throws Exception
     */
    public static void main(String[] args) throws Exception {
        ClassPathXmlApplicationContext injector = new ClassPathXmlApplicationContext("classpath:META-INF/spring/eventBusContext.xml");

        IStreamingEventBus streamingEventBus = injector.getBean("eventBus", DefaultStreamingBus.class);

        ArrayList<ModernMajorGeneralMessage> streamMessages = new ArrayList<ModernMajorGeneralMessage>();
        streamMessages.add(new ModernMajorGeneralMessage("I am "));
        streamMessages.add(new ModernMajorGeneralMessage("the very "));
        streamMessages.add(new ModernMajorGeneralMessage("model of "));
        streamMessages.add(new ModernMajorGeneralMessage("a Modern "));
        streamMessages.add(new ModernMajorGeneralMessage("Major-General, "));
        streamMessages.add(new ModernMajorGeneralMessage("I've information "));
        streamMessages.add(new ModernMajorGeneralMessage("vegetable, animal, "));
        streamMessages.add(new ModernMajorGeneralMessage("and mineral, "));
        streamMessages.add(new ModernMajorGeneralMessage("I know the kings of England, "));
        streamMessages.add(new ModernMajorGeneralMessage("and I quote the fights historical, "));
        streamMessages.add(new ModernMajorGeneralMessage("From Marathon to Waterloo, "));
        streamMessages.add(new ModernMajorGeneralMessage("in order categorical; "));


        IEventStream stream = streamingEventBus.createStream(ModernMajorGeneralMessage.class.getCanonicalName());
        stream.setBatchLimit(2);

        for (ModernMajorGeneralMessage message : streamMessages) {
            stream.publish(message);
        }
        stream.dispose();

        System.exit(0);
    }
}

File: dotnet/examples/amp.examples.streaming.publisher/StreamingPublisher.cs
using amp.eventing.streaming;
using amp.examples.streaming.common;
using cmf.eventing.patterns.streaming;
using Spring.Context;
using Spring.Context.Support;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace amp.examples.streaming.publisher
{
    public class StreamingPublisher
    {
        public static void Main(string[] args)
        {
            IApplicationContext springContainer = ContextRegistry.GetContext();

            IStandardStreamingEventBus streamingEventBus = springContainer.GetObject(
                typeof(IStandardStreamingEventBus).FullName)
                as IStandardStreamingEventBus;

            IList<ModernMajorGeneralMessage> streamMessages = new List<ModernMajorGeneralMessage>();
            streamMessages.Add(new ModernMajorGeneralMessage("I am "));
            streamMessages.Add(new ModernMajorGeneralMessage("the very "));
            streamMessages.Add(new ModernMajorGeneralMessage("model of "));
            streamMessages.Add(new ModernMajorGeneralMessage("a Modern "));
            streamMessages.Add(new ModernMajorGeneralMessage("Major-General "));
            streamMessages.Add(new ModernMajorGeneralMessage("I've information "));
            streamMessages.Add(new ModernMajorGeneralMessage("vegetable, animal, "));
            streamMessages.Add(new ModernMajorGeneralMessage("and mineral, "));
            streamMessages.Add(new ModernMajorGeneralMessage("I know the kings of England, "));
            streamMessages.Add(new ModernMajorGeneralMessage("and I quote the fights historical, "));
            streamMessages.Add(new ModernMajorGeneralMessage("From Marathon to Waterloo, "));
            streamMessages.Add(new ModernMajorGeneralMessage("in order categorical; "));


            using (IEventStream stream = streamingEventBus.CreateStream(typeof(ModernMajorGeneralMessage).FullName))
            {
                stream.BatchLimit = 2;

                foreach (ModernMajorGeneralMessage message in streamMessages)
                {
                    stream.Publish(message);
                }
            }
            
            Environment.Exit(0);
        }
    }
}
